File: 120-三角形最小路径和.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        int n = triangle.size();
        vector<int> nums = triangle[n - 1];

        for(int i = n - 2; i > -1; i--){
            for(int j = 0; j < i + 1; j++){
                nums[j] = triangle[i][j] + min(nums[j], nums[j + 1]);
            }
        }

        return nums[0];
    }
};

File: 120-三角形最小路径和.py
class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        n = len(triangle)
        nums = triangle[n - 1]

        for i in range(n - 2, -1, -1):
            for j in range(i + 1):
                nums[j] = triangle[i][j] + min(nums[j], nums[j + 1])

        return nums[0]
